src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final double RESIZE_NUMBER = 1.5;
+    private int size;
+    private Object[] elementsData = new Object[DEFAULT_SIZE];
+
+    public void resizeLength() {
+        if (size == elementsData.length) {
+            Object[] biggerArray = new Object[(int) (elementsData.length * RESIZE_NUMBER)];
+            System.arraycopy(elementsData, 0, biggerArray, 0, size);
+            elementsData = biggerArray;
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        resizeLength();
+        elementsData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        resizeLength();
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index isn't correct");
+        } else {
+            System.arraycopy(elementsData, index, elementsData, index + 1, size - index);
+            elementsData[index] = value;
+            size++;
+        }
```

Comment:
move the check of index to the private method
```suggestion
            checkIndex(index);

            System.arraycopy(elementsData, index, elementsData, index + 1, size - index);
            elementsData[index] = value;
            size++;
```

Reply:
+